engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationProcessor.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.processing.processinstance;
+
+import io.camunda.zeebe.engine.api.TypedRecord;
+import io.camunda.zeebe.engine.processing.streamprocessor.TypedRecordProcessor;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.StateWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.TypedResponseWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.Writers;
+import io.camunda.zeebe.engine.state.KeyGenerator;
+import io.camunda.zeebe.protocol.impl.record.value.processinstance.ProcessInstanceModificationRecord;
+import io.camunda.zeebe.protocol.record.intent.ProcessInstanceModificationIntent;
+
+public final class ProcessInstanceModificationProcessor
+    implements TypedRecordProcessor<ProcessInstanceModificationRecord> {
+
+  private final StateWriter stateWriter;
+  private final TypedResponseWriter responseWriter;
+  private final KeyGenerator keyGenerator;
+
+  public ProcessInstanceModificationProcessor(
+      final Writers writers, final KeyGenerator keyGenerator) {
+    stateWriter = writers.state();
+    responseWriter = writers.response();
+    this.keyGenerator = keyGenerator;
+  }
+
+  @Override
+  public void processRecord(final TypedRecord<ProcessInstanceModificationRecord> command) {
+    final ProcessInstanceModificationRecord record = command.getValue();
+
+    stateWriter.appendFollowUpEvent(
+        keyGenerator.nextKey(), ProcessInstanceModificationIntent.MODIFIED, record);
+
+    responseWriter.writeEventOnCommand(
+        keyGenerator.nextKey(), ProcessInstanceModificationIntent.MODIFIED, record, command);
```

Comment:
❌ Please make sure to reuse the generated key.

Reply:
My bad 🙈